src/compiler/emitter.ts

```diff
@@ -862,6 +862,7 @@ export function emitFiles(resolver: EmitResolver, host: EmitHost, targetSourceFi
             inlineSources: compilerOptions.inlineSources,
             extendedDiagnostics: compilerOptions.extendedDiagnostics,
             writeBundleFileInfo: !!bundleBuildInfo,
+            preserveSourceNewlines: true,

```

Comment:
I notice that `preserveSourceNewlines` is not set here would lead to not being able to use `getLeadingLineTerminatorCount` to get the accurate line number in the below and thus setting it here. Also, I am think that this might be the desired behavior to preserve such behaviors over all files.

Reply:
`preserveSourceNewlines` is true for the printer that runs refactors, which is what we’re trying to fix. This applies to `tsc` emit, and while it might be nice if it preserved input spacing, it’s too much of a performance hit. This change should be undone, and your test should be one like [this](https://github.com/microsoft/TypeScript/pull/39280/files#diff-69327758976e35d29a55b6d986ca09c9a28f074e982f07d0495051e3cb190d05) not a conformance test. When you debug a test like that, you’ll find that `preserveSourceNewlines` is already true for you. In general, be careful not to change the emitter’s behavior when `preserveSourceNewlines` is _not_ `true`.

Reply:
See also #37814